torch/nn/parallel/distributed.py

```diff
@@ -1258,21 +1258,23 @@ def register_comm_hook(self, state: object, hook: callable):
 
                             It is locally stored by each worker
                             and shared by all the gradient tensors on the worker.
-            hook (callable): Averages gradient tensors across workers and defined as:
-                             ``hook(state: object, bucket: dist.GradBucket) -> torch.futures.Future``:
+            hook (callable): Callable with the following signature:
+                             ``hook(state: object, bucket: dist.GradBucket) -> torch.futures.Future[List[torch.tensor]]``:
```

Comment:
any reason this needs to be a list of tensors instead of a single tensor? Is this legacy requirement for SPMD?

Reply:
Yes, I think it's this way for legacy reasons where a bucket replica could've had more than 1 tensor for SPMD, though we never actually supported that. I think we even somewhere do an explicit check that this is a single element list (cc @SciPioneer )